include/flexflow/simulator.h

```diff
@@ -17,6 +17,7 @@
 
 #include "config.h"
 #include "ffconst.h"
+#include "flexflow/memory_optimization.h"
```

Comment:
We do need this include because the simulator uses things here. I tried and removing it will cause compilation error.

Reply:
I can't see any changes in the diff that would require it to be there. Maybe it should be in `simualator.cc`?

Reply:
Unfortunately, I tired again to put this include in simulator.cc but it has compile errors. We should leave it here.

Reply:
There are no changes, so that shouldn't be necessary. This is just increasing build times unnecessarily. If you can't figure it out can you show me the error you're getting?